src/main/filters/SpectralTilt.cpp

```diff
@@ -0,0 +1,390 @@
+                fUpperFrequency = DFL_UPPER_FREQUENCY;
+
+            if (fLowerFrequency >= fUpperFrequency)
+            {
+                fLowerFrequency = DFL_LOWER_FREQUENCY;
+                fUpperFrequency = DFL_UPPER_FREQUENCY;
+            }
+
+            if ((enSlopeUnit == STLT_SLOPE_UNIT_NONE) || (fSlopeNepNep == 0.0f))
+            {
+                bBypass = true;
+                bSync = true;
+                return;
+            }
+            else
+            {
+                bBypass = false;
+            }
+
+            float l_angf = 2.0f * M_PI * fLowerFrequency;
+            float u_angf = 2.0f * M_PI * fUpperFrequency;
+
+            // Exponential spacing ratio for poles.
+            float r = powf(u_angf / l_angf, 1.0f / (nOrder - 1));
+            float c_fn = bilinear_coefficient(1.0f, nSampleRate);
+            float negZero = l_angf * powf(r, -fSlopeNepNep);
+            float negPole = l_angf;
+
+            // We have nOrder bilinears. We combine them 2 by 2 to get biquads.
+
+            sFilter.begin();
+            for (size_t n = 0; n < nOrder; ++n)
+            {
+                if (n % 2 != 0)
+                    continue;
+
+                bilinear_spec_t spec_now = compute_bilinear_element(negZero, negPole);
+                negZero *= r;
+                negPole *= r;
+
+                bilinear_spec_t spec_nxt = compute_bilinear_element(negZero, negPole);
+                negZero *= r;
+                negPole *= r;
+
+                dsp::biquad_x1_t *digitalbq = sFilter.add_chain();
+                if (digitalbq == NULL)
+                    return;
+
+                dsp::f_cascade_t analogbq;
+                analogbq.t[0] = spec_now.b0 * spec_nxt.b0;
+                analogbq.t[1] = spec_now.b0 * spec_nxt.b1 + spec_now.b1 * spec_nxt.b0;
+                analogbq.t[2] = spec_now.b1 * spec_nxt.b1;
+                analogbq.b[0] = spec_now.a0 * spec_nxt.a0;
+                analogbq.b[1] = spec_now.a0 * spec_nxt.a1 + spec_now.a1 * spec_nxt.a0;
+                analogbq.b[2] = spec_now.a1 * spec_nxt.a1;
+
+                dsp::bilinear_transform_x1(digitalbq, &analogbq, c_fn, 1);
+
+                // The denominator coefficients in digitalbq will have opposite sign with respect the maths.
+                // This is correct, as this is the LSP convention.
+            }
+            sFilter.end(true);
+
+            bSync = true;
+        }
+
+        void SpectralTilt::process_add(float *dst, const float *src, size_t count)
+        {
+            if (src != NULL)
+                dsp::copy(dst, src, count);
+            else
+                dsp::fill_zero(dst, count);
+
+            if (bBypass)
+            {
+                dsp::mul_k2(dst, 2.0f, count);
+                return;
+            }
+
+            while (count > 0)
+            {
+                size_t to_do = lsp_min(count, BUF_LIM_SIZE);
+
+                sFilter.process(vBuffer, dst, to_do);
```

Comment:
I did this also to `ButterworthFilter`, `Velvet` and [NoiseGenerator](https://github.com/lsp-plugins/lsp-dsp-units/pull/2#discussion_r767609975).

Reply:
By the way, I used statements of this kind.

```c++
#define BUF_LIM_SIZE        1024u

namespace lsp
{
    namespace dspu
    {

        // 1X buffer for temporary processing.
        float vVelvetTempBuf[BUF_LIM_SIZE]{};
```
Is this what you had in mind @sadko4u ?

Reply:
No, that's not correct because `vVelvetTempBuf` is a global variable now and may cause race condition between different threads that can run the processing function.
`vVelvetTempBuf` can be isolated by allocating a separate memory chunk for each DSP processing instance (as it was done) or utilizing the thread stack for storing data (local variable inside the function) since the thread stack is isolated from another threads. So reducing `BUF_LIM_SIZE` to something like 256 elements and making the buffer a local variable inside the processing function would be good.